lib/liquid/tags/assign.rb

```diff
@@ -18,13 +18,12 @@ def self.syntax_error_translation_key
 
     attr_reader :to, :from
 
-    def initialize(tag_name, markup, options)
-      super
-      if markup =~ Syntax
+    def parse(_tokens)
+      if @markup =~ Syntax
         @to = Regexp.last_match(1)
-        @from = Variable.new(Regexp.last_match(2), options)
+        @from = Variable.new(Regexp.last_match(2), @parse_context)
```

Comment:
Nit: we could use the accessors (`parse_context`)

Reply:
I'm not sure why I have done this as `options` is also available.

Reply:
I think I discovered all these methods are accessors anyway so most don't even need to be modified.

Reply:
OK, I've fixed these all up, there are very minimal changes now.